### PR TITLE
add enter point for order ccu by film date

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: api/rest.go [hunk 1]
```diff
@@ -217,6 +217,29 @@ func CollectionContentUnitsHandler(c *gin.Context) {
 	concludeRequest(c, resp, err)
 }
 
+func CollectionContentUnitsPositionHandler(c *gin.Context) {
+	var err *HttpError
+	var resp interface{}
+	id, e := strconv.ParseInt(c.Param("id"), 10, 0)
+	if c.Request.Method == http.MethodPost {
```
Comment (edoshor) [inline api/rest.go:224]: useless check. The router does that for you.

File: api/rest.go [hunk 2]
```diff
@@ -1585,6 +1608,29 @@ func handleUpdateCollection(cp utils.ContextProvider, exec boil.Executor, c *Par
 	return handleGetCollection(cp, exec, c.ID)
 }
 
+func handleCollectionContentUnitsPosition(exec boil.Executor, id int64) (*events.Event, *HttpError) {
+	collection, err := models.FindCollection(exec, id)
+	event := events.CollectionContentUnitsChangeEvent(collection)
+	if err != nil {
+		return &event, NewInternalError(err)
+	}
+	ccus, errF := models.CollectionsContentUnits(exec,
```
Comment (edoshor) [inline api/rest.go:1617]: I had this sql snippet used manually many times. Maybe use it here ?
It does the whole thing in 1 DB call.

```sql
UPDATE collections_content_units
SET position = s.row_number
FROM (select cu.id, cu.properties ->> 'film_date', ccu.position, ROW_NUMBER()
    OVER (
      ORDER BY cu.properties ->> 'film_date') as row_number
      from collections_content_units ccu
             inner join content_units cu on ccu.content_unit_id = cu.id and ccu.collection_id = ?
      order by cu.properties ->> 'film_date') AS s
WHERE content_unit_id = s.id
  and collection_id = ?;
```

File: api/rest.go [hunk 3]
```diff
@@ -217,6 +217,29 @@ func CollectionContentUnitsHandler(c *gin.Context) {
 	concludeRequest(c, resp, err)
 }
 
+func CollectionContentUnitsPositionHandler(c *gin.Context) {
+	var err *HttpError
+	var resp interface{}
+	id, e := strconv.ParseInt(c.Param("id"), 10, 0)
+	if c.Request.Method == http.MethodPost {
+
+		if e != nil {
```
Comment (edoshor) [inline api/rest.go:226]: error checking / handling in golang should be next to where it happens. Not later on in the code

**Review by edoshor — CHANGES_REQUESTED** [review]

File: api/rest.go (hunk 1 above)
Comment (davgur) [reply api/rest.go]: fixed

File: api/rest.go (hunk 3 above)
Comment (davgur) [reply api/rest.go]: fixed

File: api/rest.go (hunk 2 above)
Comment (davgur) [reply api/rest.go]: fixed
I was think that if it's on one transaction it same as single request

**Pull request merged**